src/CMakeLists.txt

```
@@ -339,7 +339,11 @@ function(test_func_impl TEST_NAME NUM_PROCS)
       endif()
       set(TEST_STR ${VALGRIND} ${ARGN} ${CTEST_MEMPOOL_ARG})
     endif()
-    add_test(NAME ${TEST_NAME} COMMAND ${TEST_STR})
+
+    list(POP_FRONT ARGN TEST_EXEC)
+    string(REPLACE "./" "" TEST_EXEC ${TEST_EXEC})
+
+    add_test(NAME ${TEST_NAME} COMMAND ${CMAKE_COMMAND} -E env $<TARGET_FILE:${TEST_EXEC}> ${VALGRIND} ${ARGN} ${CTEST_MEMPOOL_ARG}) # https://cmake.org/cmake/help/latest/prop_test/WILL_FAIL.html#prop_test:WILL_FAIL
```

The following change fixed the parallel tests for me:

```suggestion

    # need to run as a cmake script to capture assert and other 'system failures'
    # https://cmake.org/cmake/help/latest/prop_test/WILL_FAIL.html#prop_test:WILL_FAIL
    add_test(NAME ${TEST_NAME} COMMAND ${CMAKE_COMMAND} -E env ${TEST_STR})

```